README.md

```diff
@@ -1,6 +1,94 @@
+[![Build Status](https://dev.azure.com/cncf/strimzi/_apis/build/status/kafka-kubernetes-config-provider?branchName=main)](https://dev.azure.com/cncf/strimzi/_build/latest?definitionId=32&branchName=main)
+[![GitHub release](https://img.shields.io/github/release/strimzi/kafka-kubernetes-config-provider.svg)](https://github.com/strimzi/kafka-kubernetes-config-provider/releases/latest)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.strimzi/kafka-kubernetes-config-provider/badge.svg)](https://maven-badges.herokuapp.com/maven-central/io.strimzi/kafka-kubernetes-config-provider)
 [![License](https://img.shields.io/badge/license-Apache--2.0-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)
 [![Twitter Follow](https://img.shields.io/twitter/follow/strimziio.svg?style=social&label=Follow&style=for-the-badge)](https://twitter.com/strimziio)
 
-# Strimzi Template repository
+# Kubernetes Configuration Provider for Apache Kafka
 
-This repository is currently empty
+Apache Kafka supports pluggable configuration providers which can load configuration data from external sources.
+This configuration provider can be used to load data from Kubernetes Secrets and Config Maps.
+It can be used in all Kafka components.
+One of the example use-cases is to load certificates or JAAS configuration from Kubernetes Secrets.
+
+## Adding the Kubernetes Configuration Provider to Apache Kafka clients
+
+You can add Kubernetes Configuration Provider as any other Java dependency using Maven or any other build tool.
+For example:
+
+```xml
+<dependency>
+    <groupId>io.strimzi</groupId>
+    <artifactId>kafka-kubernetes-config-provider</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+## Adding the Kubernetes Configuration Provider to Apache Kafka server components
+
+To add Kubernetes Configuration Provider to the Apache Kafka server distribution, you can download the ZIP file and unpack it into Kafka's `libs` directory.
```

Comment:
Well, this would be always valid for users using it in their own components without outside of Strimzi. So I think this will be still valid.

TBH, I haven't thought about adding to our images. I saw it more as a clients thing. But we can do it ... it is just the RBAC stuff will make it a bit weird I guess, but the operands have stable service accounts, so they can just create some additional RoleBindings very easily. It could be maybe better alternative then mounting them through the volumes / environment variables since you do not need any restarts and you can basically do it as part of an connector.